src/PackagePrivate/InMemoryTermIdsAcquirer.php

```diff
@@ -30,4 +30,23 @@ public function acquireTermIds( array $termsArray ): array {
 		return $ids;
 	}
 
+	public function cleanTerms( array $termInLangIds ) {
+		$termInLangIdsAsKeys = array_flip( $termInLangIds );
+		foreach ( $this->terms as $type => &$termsOfType ) {
+			foreach ( $termsOfType as $lang => &$termsOfLang ) {
+				foreach ( $termsOfLang as $term => $id ) {
+					if ( array_key_exists( $id, $termInLangIdsAsKeys ) ) {
```

Comment:
PHPMD would be very mad if it where included in the CS. Lots of indent. I'd have split this into multiple functions to reduce scope of the many variables. Local and minor issue though, fine with merging this.